src/shared/parse-argument.c

```diff
@@ -10,6 +10,25 @@
 
 /* All functions in this file emit warnigs. */
 
+int parse_boolean_argument(const char *optname, const char *s, bool *ret) {
+        int r;
+
+        /* Returns the result through *ret and the return value. */
+
+        if (s) {
+                r = parse_boolean(s);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to parse %s= argument.", optname);
```

Comment:
why not:

```c
return log_error_errno(r, "Failed to parse %s= argument: %s", optname, s);
```

(i.e. also say what string precisely failed parsing. people might use the same option multiple times, who knows)